src/client/actions/TypingStart.js

```diff
@@ -7,7 +7,7 @@ const textBasedChannelTypes = ['dm', 'text', 'news'];
 class TypingStart extends Action {
   handle(data) {
     const channel = this.getChannel(data);
-    if (!textBasedChannelTypes.includes(channel.type)) {
+    if (channel && !textBasedChannelTypes.includes(channel.type)) {
```

Comment:
Shouldn't we exit early if no channel is present?

Reply:
Done. Though that way we don't cache any member data anymore if it's there. Depends on if we want to cache everything we possibly can or are more lax.

Reply:
I mean... if we don't have the channel but we cache the member...we still don't emit any events, so I believe it's probably best to exit early. That, of course, is open for discussions tho